File: dashboard/base_api_handler.py
import requests


def request_handler(get_domain_url, endpoint, request_type, payload, request):
    user_id = '' if 'user_id' not in request.COOKIES else request.COOKIES['user_id']
    access_token = '' if 'access_token' not in request.COOKIES else request.COOKIES['access_token']
    headers = {
        'accept': 'application/json',
        'Authorization': 'Bearer '+access_token+'',
    }
    url = ''+get_domain_url+'/rest_api/'+endpoint+'?user_id='+user_id+''
    if request_type == 'post':
        response = requests.post(url, data=payload, headers=headers)
    elif request_type == 'get':
        response = requests.get(url, data=payload, headers=headers)
    status_code = response.status_code
    json_data = response.json()
    if status_code == 200:
        return json_data

File: static/js/api/base_api_handler.js
function request(request_type, payload, endpoint) {
  if (user_id == "") {
    user_id = session_user_id;
  }

    return $.ajax({
      url: '' + domain_url + '/rest_api/' + endpoint+ '?user_id='+user_id,
      type: request_type,
      dataType: "json",
      data: payload,
      headers: {
        "accept": "application/json",
        'Authorization': 'Bearer '+access_token+'',
        "Content-Type": "application/x-www-form-urlencoded"
      },
    });
  }
  

  function removeVehicleArray(jsonStr) {
    const jsonObj = JSON.parse(jsonStr);
    const vehicleArray = jsonObj.vehicleArray; // Extract the content of vehicleArray
    delete jsonObj.vehicleArray; // Delete the vehicleArray key
    Object.assign(jsonObj, vehicleArray[0]); // Merge the content of vehicleArray into jsonObj
    return JSON.stringify(jsonObj);
  }


  function request_vehicle(request_type, payload, endpoint) {

   const payload_json = removeVehicleArray(payload);
    alert("test")
    alert(user_id)
    console.log(payload_json)
    if (user_id == "") {
      user_id = session_user_id;
    }
    alert(user_id)
      return $.ajax({
        url: '' + domain_url + '/rest_api/' + endpoint+ '?company_id='+user_id,
        type: request_type,
        dataType: "json",
        data: payload_json,
        headers: {
          "accept": "application/json",
          'Authorization': 'Bearer '+access_token+'',
          "Content-Type": "application/json"
        },
      });
    }
    